drivers/interrupt_controller/intc_mtk_adsp.c

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2023 The ChromiumOS Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+
+#define DT_DRV_COMPAT mediatek_adsp_intc
+
+struct intc_mtk_cfg {
+	uint32_t xtensa_irq;
+	uint32_t irq_mask;
+	uint32_t sw_isr_off;
+	volatile uint32_t *enable_reg;
+	volatile uint32_t *status_reg;
+};
+
+bool intc_mtk_adsp_get_enable(const struct device *dev, int irq)
```

Comment:
There might be a reason this is in the global scope, but if not, I would suggest keeping this and other functions below static.
```suggestion
static bool intc_mtk_adsp_get_enable(const struct device *dev, int irq)
```

Reply:
Yeah, these need to be called out of the SOC-side irq_en/disable() implementation (in soc/.../irq.c in this PR).  There's no generic mechanism for that.  There was some discussion about a new API (c.f. @ycsin @bjarki-trackunit) a few months back that would tie things together, but it's a big task and everyone has different requirements.